pkg/plugin/aggregation/server.go

```diff
@@ -18,37 +18,32 @@ package aggregation
 
 import (
 	"fmt"
-	"net"
 	"net/http"
 	"strings"
 
-	"github.com/sirupsen/logrus"
 	"github.com/heptio/sonobuoy/pkg/plugin"
-	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
 // Server is a net/http server that can handle API requests for aggregation of
 // results from nodes, sending them back over the Results channel
-type Server struct {
-	// BindAddr is the address for the HTTP server to listen on, eg. 0.0.0.0:8080
-	BindAddr string
+type AggregationHandler struct {
+	mux *http.ServeMux
```

Comment:
If the struct gets embedded directly then the function online 66 can disappear. Also the calls to `handler.mux.Handle` would become `handler.Handle` which seems cleaner.

Reply:
@chuckha `http.ServeMux` has an initalisation function that returns a pointer to the struct, the documentation doesn't make the usual guarantees about "a zero-value struct is valid." Looking at the code, there's a map in there which usually needs to get initialised. How does that usually work?

Reply:
The ServeMux would still need to be initialized, but the initialization for Aggregation server would look like:

```
handler := &AggregationHandler{
ServeMux: http.NewServeMux(), 
ResultsCallback: resultsCallback,
}
```
instead of defining each field. Omitting field names is also an option.

Reply:
https://stackoverflow.com/questions/12537496/initialize-embedded-struct-in-go lol, just needed to google it 😅